x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts

```diff
@@ -282,16 +287,14 @@ export class SecureSpacesClientWrapper implements ISpacesClient {
               // This object exists in All Spaces and its `namespaces` field isn't going to change; there's nothing to audit
               return;
             }
-            this.auditLogger.log(
-              savedObjectEvent({
-                action: isOnlySpace
-                  ? SavedObjectAction.DELETE
-                  : SavedObjectAction.UPDATE_OBJECTS_SPACES,
-                outcome: 'unknown',
-                savedObject: { type: savedObject.type, id: savedObject.id },
-                deleteFromSpaces: [id],
-              })
-            );
+            this.securityExtension!.addAuditEvent({
+              action: isOnlySpace
+                ? SavedObjectAction.DELETE
+                : SavedObjectAction.UPDATE_OBJECTS_SPACES,
+              outcome: 'unknown',
+              savedObject: { type: savedObject.type, id: savedObject.id },
+              ...(!isOnlySpace && { deleteFromSpaces: [id] }),
+            });
```

Comment:
The old audit event included the `deleteFromSpaces` field whether the object was being updated or deleted.

Now, if the object is being _deleted_, we omit the `deleteFromSpaces` field.